### PR TITLE
Playlist page & add to playlist

### PR DESCRIPTION
**Pull request by aayushjain-code**

### CODE REVIEW

File: src/components/ExplorePage/exploreCard.js [hunk 1]
```diff
@@ -1,22 +1,99 @@
...
-const ExploreCard = ({ item }) => {
+import { usePlaylist } from "../../context/playlistContext";
+import { CardVideoPlaylist } from "../HomePage/cardVideoPlaylist";
+
+const FeaturedCard = ({ item }) => {
   const [dropdown, setDropdown] = useState(false);
+  const [isSavetoPlaylistClicked, setIsSavetoPlaylistClicked] = useState(false);
+  const [clickedCreateNewPlaylist, setClickedCreateNewPlaylist] =
+    useState(false);
+  // const [isMoreOptions, setIsMoreOptions] = useState(false);
```
Comment (b0llu) [inline src/components/ExplorePage/exploreCard.js:14]: remove this commented state if your not going to use it in future

File: src/components/ExplorePage/exploreCard.js [hunk 2]
```diff
@@ -1,22 +1,99 @@
...
+    getAllPlaylists();
+  }, [
+    clickedCreateNewPlaylist,
+    addVideoToplaylistLoading,
+    removeVideoFromPlaylistLoading,
+  ]);
+
+  // useEffect(() => {
+  //   console.log("playlists:", playlists);
+  // });
```
Comment (b0llu) [inline src/components/ExplorePage/exploreCard.js:65]: Remove this useEffect if not going to use it in future

File: src/components/HomePage/FeaturedCard.js [hunk 3]
```diff
@@ -1,13 +1,82 @@
...
 import { useHistory } from "../../context/historyContext";
 import { trimHeading } from "../../utils/Utilities/utilFunctions";
+import { usePlaylist } from "../../context/playlistContext";
+import { CardVideoPlaylist } from "./cardVideoPlaylist";
 const FeaturedCard = ({ item }) => {
   const [dropdown, setDropdown] = useState(false);
+  const [isSavetoPlaylistClicked, setIsSavetoPlaylistClicked] = useState(false);
+  const [clickedCreateNewPlaylist, setClickedCreateNewPlaylist] =
+    useState(false);
+  // const [isMoreOptions, setIsMoreOptions] = useState(false);
```
Comment (b0llu) [inline src/components/HomePage/FeaturedCard.js:14]: Remove if not going to use in future

**Review by b0llu — APPROVED** [review]
Your code looks good just a suggestion remove commented code if your not going to use it. You have lot of code commented out which is not a great thing for PR's. You can merge this

File: src/components/ExplorePage/exploreCard.js [hunk 4]
```diff
@@ -43,30 +120,116 @@ const ExploreCard = ({ item }) => {
...
+                      key={item._id}
+                    />
+                  );
+                })}
+            </div>
+
+            {/* <div className="playlist-second-div">
+              <input type="checkbox" className="playlist-checkbox" />
+              <span className="playlist-title">Title</span>
+            </div> */}
```
Comment (Developers-codz) [inline src/components/ExplorePage/exploreCard.js:182]: remove commented code if not needed

File: src/components/PlaylistPage/CardVideo.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+import { usePlaylist } from "../../context/playlistContext";
+import { useWatchLater } from "../../context/watchLaterContext";
+
+import { trimHeading } from "../../utils/Utilities/utilFunctions";
+
+const CardVideo = ({ item }) => {
+  const [isSavetoPlaylistClicked, setIsSavetoPlaylistClicked] = useState(false);
+  const [clickedCreateNewPlaylist, setClickedCreateNewPlaylist] =
+    useState(false);
+  // const [isMoreOptions, setIsMoreOptions] = useState(false);
```
Comment (Developers-codz) [inline src/components/PlaylistPage/CardVideo.js:13]: from here as well

File: src/components/PlaylistPage/CardVideo.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    getAllPlaylists();
+  }, [
+    clickedCreateNewPlaylist,
+    addVideoToplaylistLoading,
+    removeVideoFromPlaylistLoading,
+  ]);
+
+  // useEffect(() => {
+  //   console.log(playlists);
+  // });
```
Comment (Developers-codz) [inline src/components/PlaylistPage/CardVideo.js:64]: here

File: src/components/PlaylistPage/SingleVideoFeaturedCard.js [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+    getAllPlaylists();
+  }, [
+    clickedCreateNewPlaylist,
+    addVideoToplaylistLoading,
+    removeVideoFromPlaylistLoading,
+  ]);
+
+  // useEffect(() => {
+  //   console.log("playlists:", playlists);
+  // });
```
Comment (Developers-codz) [inline src/components/PlaylistPage/SingleVideoFeaturedCard.js:65]: here

File: src/components/PlaylistPage/SingleVideoFeaturedCard.js [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+  return (
+    <div className="category" key={item._id}>
+      <div className="box-container">
+        <div className="box">
+          <Link
+            to={`/singlevideo/${item._id}`}
+            onClick={() => addVideoToHistory(item)}
+          >
+            <div className="image">
+              <img src={item.imageUrl} alt={item.title} loading="lazy" />
```
Comment (Developers-codz) [inline src/components/PlaylistPage/SingleVideoFeaturedCard.js:106]: Appreciation- used loading lazy

**Review by Developers-codz — APPROVED** [review]
UI Looks Amazingggg
Code is also well structured
Just small suggestion- keep PR small, It will be easy for reviewers to review

File: src/components/ExplorePage/exploreCard.js [hunk 9]
```diff
@@ -1,22 +1,99 @@
...
+
+  useEffect(() => {
+    getAllPlaylists();
+  }, [
+    clickedCreateNewPlaylist,
+    addVideoToplaylistLoading,
+    removeVideoFromPlaylistLoading,
+  ]);
+
+  // useEffect(() => {
```
Comment (MuskanChhatrasal) [inline src/components/ExplorePage/exploreCard.js:63]: Remove this if not in use.

File: src/components/ExplorePage/exploreCard.js [hunk 10]
```diff
@@ -43,30 +120,116 @@ const ExploreCard = ({ item }) => {
...
                     <i className="fa-regular fa-clock dropIcon"></i>
                     Add to watch later
                   </li>
                 )}
-                <li>
+
+                <li
+                  onClick={() => {
+                    setIsSavetoPlaylistClicked(true);
+                    // setDropdown(!dropdown);
```
Comment (MuskanChhatrasal) [inline src/components/ExplorePage/exploreCard.js:140]: Here also remove commented code

**Review by MuskanChhatrasal — APPROVED** [review]
Amazing UI with the cleaner and readable code. Just one suggestion remove all the commented code then it will be more readable and cleaner

**Pull request merged**